fix: Native AbortController compatibility for Node.js 18+ environments

rmlearney-digicatapult opened the pull request:
**Problem**
Credo-ts now requires Node >= 18. We are currently using the `abort-controller` polyfill (v3.0.0) which creates compatibility issues with Node.js 18+ native `fetch()` implementation. 

The polyfill's `AbortSignal` instances fail `instanceof` checks against native `AbortSignal`, causing runtime errors.

```sh
TypeError: Expected signal to be an instance of AbortSignal
    at Object.fetch (node:internal/deps/undici/undici:11413:13)
    at HttpOutboundTransport.sendMessage (/path/to/credo-ts/packages/core/src/transport/HttpOutboundTransport.ts:78:24)
    at async MessageSender.sendMessage (/path/to/credo-ts/packages/core/src/agent/MessageSender.ts:156:7)
```

This issue affects HTTP operations in:

- DIDComm message transport in `HttpOutboundTransport`
- Shortened invitation URL fetching in `parseInvitation`

**Solution**
1. Native AbortController Migration
- Removed abort-controller polyfill from core package, uses Node.js/Browser native AbortController (Node >= 18, React Native >= 0.60 have `AbortController` support)
- Removed eslint restrictions that enforced polyfill usage


2. Refactoring
- Uses `fetchWithTimeout` utility to replace duplicate timeout logic and have single source of truth for timeout-based fetch operations
- Preserved all HTTP request characteristics (methods, headers, timeouts, error handling)

3. Backward Compatibility
- All existing APIs unchanged
- All HTTP operations work identically
- No changes to supported environments

TimoGlastra reviewed (approved):
I was just running into this issue today with @berendsliedrecht. We assumed we had already backported this, but seems not. Thanks!

TimoGlastra commented:
can you update the lockfile @rmlearney-digicatapult?

rmlearney-digicatapult commented:
Gah. Messed up the branch.

rmlearney-digicatapult committed:
feat: migrate to native AbortController for Node.js 18+ compatibility

- Remove abort-controller polyfill dependency from core package
- Update all imports to use native AbortController instead of polyfill
- Refactor HttpOutboundTransport to use fetchWithTimeout utility
- Update parseInvitation to use fetchWithTimeout utility
- Remove ESLint restriction against global AbortController
- Preserve all existing HTTP request characteristics and error handling

This fixes TypeError when polyfill AbortSignal instances fail instanceof
checks against native AbortSignal in Node.js 18+ fetch() implementation.

Signed-off-by: rmlearney-digicatapult <robert.learney@digicatapult.org.uk>

rmlearney-digicatapult committed:
Update signal as NonNullable

Signed-off-by: rmlearney-digicatapult <robert.learney@digicatapult.org.uk>

TimoGlastra commented:
It seems prettier is still failing

rmlearney-digicatapult commented:
Odd...seems to be the webvh README. Will take a look

genaris committed:
style: fix prettier

Signed-off-by: Ariel Gentile <gentilester@gmail.com>

genaris committed:
docs(changeset): fix: Native AbortController compatibility for Node.JS 18+ environments

Signed-off-by: Ariel Gentile <gentilester@gmail.com>
